### ISSUE TITLE
Unable to create bgp sessions

### ISSUE THREAD

**Issue by JRBoniface** [terraform-google-modules/terraform-google-vpn#74]
### TL;DR

Cloud VPN Gateway, Peer VPN Gateway and tunnel created successfully however an error occurs creating the BGP session for the tunenl.

`Error: Error creating RouterBgpPeer: googleapi: Error 400: Invalid value for field 'resource.bgpPeers[2]': '{  "name": "testvpn2",  "interfaceName": "if-testvpn2...'. BGP peer "testvpn2" associated with multiple interfaces: if-tec-manage-prod-crp-euwe2-routerprivate-3d8f-qavpn2-1; if-tec-manage-prod-crp-euwe2-routerprivate-3d8f-testvpn2, invalid`

### Expected behavior

To have configured the Bgp session with the defined values

### Observed behavior

Error in terraform 

Error: Error creating RouterBgpPeer: googleapi: Error 400: Invalid value for field 'resource.bgpPeers[2]': '{  "name": "testvpn2",  "interfaceName": "if-testvpn2...'. BGP peer "testvpn2" associated with multiple interfaces: if-tec-manage-prod-crp-euwe2-routerprivate-3d8f-qavpn2-1; if-tec-manage-prod-crp-euwe2-routerprivate-3d8f-testvpn2, invalid

### Terraform Configuration

```hcl
resource "google_compute_ha_vpn_gateway" "ha_gateway" {
  for_each = toset(var.ha_vpn_gateways)

  name       = "${local.standardized_external_vpn_gateway}-${each.value}"
  region     = var.region
  network    = var.network
  depends_on = [var.network]
}

resource "google_compute_external_vpn_gateway" "external_gateway" {
  for_each        = toset(keys({ for k, v in var.peer_external_gateway : k => v }))
  name            = "${local.standardized_external_vpn_gateway}-${each.key}"
  redundancy_type = var.peer_external_gateway[each.value]["redundancy_type"]
  description     = "An externally managed VPN gateway"

  dynamic "interface" {
    for_each = var.peer_external_gateway[each.value]["interfaces"]
    content {
      id         = interface.value.id
      ip_address = interface.value.ip_address
    }
  }
  timeouts {}

  lifecycle {
    ignore_changes = [description]
  }
}

resource "google_compute_vpn_tunnel" "tunnel_ha" {
  for_each = toset(keys({ for k, v in var.tunnels : k => v }))

  name          = "${local.standardized_vpn_tunnel}-${each.key}"
  project       = var.project
  region        = var.region
  shared_secret = data.google_secret_manager_secret_version.vpn_ha[var.tunnels[each.value]["shared_secret"]].secret_data

  local_traffic_selector  = null
  remote_traffic_selector = null

  vpn_gateway = google_compute_ha_vpn_gateway.ha_gateway[var.tunnels[each.value]["vpn_gateway_name"]].id

  peer_external_gateway           = google_compute_external_vpn_gateway.external_gateway[each.key].id
  peer_external_gateway_interface = var.tunnels[each.value]["peer_external_gateway_interface"]
  router                          = var.router
  vpn_gateway_interface           = var.tunnels[each.value]["vpn_gateway_interface"]

  depends_on = [var.router, google_compute_ha_vpn_gateway.ha_gateway, google_compute_external_vpn_gateway.external_gateway]

  lifecycle {
    ignore_changes = [shared_secret]
  }
}

resource "google_compute_router_interface" "router_interface" {
  for_each = toset(keys({ for k, v in var.tunnels : k => v }))

  name   = "if-${local.standardized_router_interface}-${each.key}"
  router = var.router_name
  region = var.region

  vpn_tunnel = google_compute_vpn_tunnel.tunnel_ha[each.key].self_link

  lifecycle {
    ignore_changes = [ip_range]
  }

  timeouts {}

  depends_on = [var.router, google_compute_vpn_tunnel.tunnel_ha]
}

resource "google_compute_router_peer" "router_peers" {
  for_each = toset(keys({ for k, v in var.tunnels : k => v }))

  name     = "${local.standardized_router_peer}-${each.key}"
  router   = var.router_name
  region   = var.region
  peer_asn = lookup(var.tunnels[each.value].bgp_peer, "asn", "")

  advertised_route_priority = 0

  peer_ip_address = lookup(var.tunnels[each.value].bgp_peer, "address", "")
  ip_address      = lookup(var.tunnels[each.value], "bgp_session_ip", "")
  interface       = google_compute_router_interface.router_interface[each.key].name

  lifecycle {
    ignore_changes = [peer_ip_address]
  }

  depends_on = [var.router, google_compute_router_interface.router_interface]
}



variable "peer_external_gateway" {
  description = "Configuration of an external VPN gateway to which this VPN is connected."
  type = map(object({
    redundancy_type = string
    interfaces = list(object({
      id         = number
      ip_address = string
    }))
  }))
  default = {}
}

variable "tunnels" {
  description = "VPN tunnel configurations, bgp_peer_options is usually null."
  type = map(object({
    bgp_peer = object({
      address = string
      asn     = number
    })
    bgp_peer_options = object({
      advertise_groups    = list(string)
      advertise_ip_ranges = map(string)
      advertise_mode      = string
      route_priority      = number
    })
    bgp_session_ip                  = string
    bgp_session_range               = string
    ike_version                     = number
    vpn_gateway_name                = string
    vpn_gateway_interface           = number
    peer_external_gateway_interface = number
    shared_secret                   = string
  }))
  default = {}
}

variable "ha_vpn_gateways" {
  type        = list(any)
  description = "Specifies the name of each cloud vpn gateway to be created in gcp"
  default     = []
}
```

### Terraform Version

```sh
Terraform v0.14.11
+ provider registry.terraform.io/hashicorp/google v3.90.1
+ provider registry.terraform.io/hashicorp/google-beta v3.90.1
+ provider registry.terraform.io/hashicorp/kubernetes v2.6.1
```

**Comment by jonathanrboniface** [comment]
Having reviewed the configuration further there was a problem with the input values not the configuration of the module defined above so this is not an issue and should've closed.
